presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowStatsRewriter.java

```diff
@@ -0,0 +1,227 @@
+
+        if (tableLayouts.isEmpty()) {
+            return Optional.empty();
+        }
+
+        return Optional.of(tableLayouts.get(0).getLayout().getHandle());
+    }
+
+    private Node createEmptyStatsTable(QualifiedName tableName)
+    {
+        return simpleQuery(selectList(unaliasedName("column_name")), aliased(new Values(ImmutableList.of()), "table_stats_for_" + tableName, ImmutableList.of("column_name")));
+    }
+
+    private List<TableLayoutResult> getLayouts(TableScanNode node, Expression predicate, Map<Symbol, Type> types)
+    {
+        DomainTranslator.ExtractionResult decomposedPredicate = DomainTranslator.fromPredicate(
+                metadata,
+                session,
+                predicate,
+                types);
+
+        TupleDomain<ColumnHandle> simplifiedConstraint = decomposedPredicate.getTupleDomain()
+                .transform(node.getAssignments()::get)
+                .intersect(node.getCurrentConstraint());
+
+        return metadata.getLayouts(
+                session, node.getTable(),
+                new Constraint<>(simplifiedConstraint, bindings -> true),
+                Optional.of(ImmutableSet.copyOf(node.getAssignments().values())));
+    }
+
+    private Map<ColumnHandle, String> getStatisticsColumnNames(TableStatistics statistics, ShowStats node, QualifiedName tableName)
+    {
+        ImmutableMap.Builder<ColumnHandle, String> resultBuilder = ImmutableMap.builder();
+        TableHandle tableHandle = getTableHandle(node, tableName);
+
+        for (ColumnHandle column : statistics.getColumnStatistics().keySet()) {
+            resultBuilder.put(column, metadata.getColumnMetadata(session, tableHandle, column).getName());
+        }
+
+        return resultBuilder.build();
+    }
+
+    private TableHandle getTableHandle(ShowStats node, QualifiedName table)
+    {
+        QualifiedObjectName qualifiedTableName = createQualifiedObjectName(session, node, table);
+        return metadata.getTableHandle(session, qualifiedTableName)
+                .orElseThrow(() -> new SemanticException(MISSING_TABLE, node, "Table %s not found", table));
+    }
+
+    private static List<String> findUniqueStatisticsNames(TableStatistics tableStatistics)
+    {
+        TreeSet<String> statisticsKeys = new TreeSet<>();
+        statisticsKeys.addAll(tableStatistics.getTableStatistics().keySet());
+        for (ColumnStatistics columnStats : tableStatistics.getColumnStatistics().values()) {
+            statisticsKeys.addAll(columnStats.getStatistics().keySet());
+        }
+        return unmodifiableList(new ArrayList(statisticsKeys));
+    }
+
+    private static Optional<TableScanNode> findTableScanNode(Plan plan)
+    {
+        List<TableScanNode> scanNodes = PlanNodeSearcher.searchFrom(plan.getRoot()).where(TableScanNode.class::isInstance).findAll();
+        checkState(scanNodes.size() <= 1, "There must be max of 1 table scan nodes in SHOW STATS SELECT clause");
```

Comment:
static import
I would add `getOnlyElement` to `PlanNodeSearcher` (like https://google.github.io/guava/releases/snapshot/api/docs/com/google/common/collect/Iterables.html#getOnlyElement-java.lang.Iterable-T-)